app/Domains/Vault/ManageReports/Web/ViewHelpers/ReportImportantDateSummaryIndexViewHelper.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace App\Domains\Vault\ManageReports\Web\ViewHelpers;
+
+use App\Helpers\ContactCardHelper;
+use App\Helpers\DateHelper;
+use App\Helpers\ImportantDateHelper;
+use App\Models\Contact;
+use App\Models\ContactImportantDate;
+use App\Models\User;
+use App\Models\Vault;
+use Carbon\Carbon;
+use Illuminate\Support\Collection;
+
+class ReportImportantDateSummaryIndexViewHelper
+{
+    /**
+     * Get all the important dates for a given contact.
+     * This screen will list all the important dates for the next 12 months.
+     *
+     * @param  Vault  $vault
+     * @param  User  $user
+     * @return Collection
+     */
+    public static function data(Vault $vault, User $user): Collection
+    {
+        $contactsInVault = $vault->contacts->pluck('id')->toArray();
+        $importantDates = ContactImportantDate::whereIn('contact_id', $contactsInVault)
+            ->with('contact')
+            ->orderBy('month', 'asc')
+            ->orderBy('day', 'asc')
+            ->get();
+
+        // create a loop looping over the next 12 months
+        $currentDate = Carbon::now()->copy();
+        $monthsCollection = collect();
+        for ($month = 0; $month < 12; $month++) {
+            $date = $currentDate->copy();
+            $date->addMonths($month);
+
+            $importantDatesCollection = collect();
+            foreach ($importantDates as $importantDate) {
+                if ($importantDate->month === $date->month) {
```

Comment:
I don't understand. So you have 2 loops, but it actually only shows the important dates of the current month right? You could query the important dates of the month directly, and only iterate once over it?

Reply:
No, because that would do 12 queries instead of one.
At the moment, I do one query (which could, indeed, contains many items) then I iterate over a collection, not over another query.